File: app/src/main/java/com/disruption/cookcentral/CookCentralApplication.java
package com.disruption.cookcentral;

import androidx.multidex.MultiDexApplication;
import androidx.work.Constraints;
import androidx.work.ExistingPeriodicWorkPolicy;
import androidx.work.NetworkType;
import androidx.work.PeriodicWorkRequest;
import androidx.work.WorkManager;

import com.disruption.cookcentral.work.RandomRecipeWorker;

import java.util.concurrent.TimeUnit;

public class CookCentralApplication extends MultiDexApplication {

    @Override
    public void onCreate() {
        super.onCreate();
        setUpRecipeWork();
    }

    private void setUpRecipeWork() {
        Constraints constraints = new Constraints.Builder()
                .setRequiredNetworkType(NetworkType.CONNECTED)
                .setRequiresBatteryNotLow(true)
                .build();

        PeriodicWorkRequest request =
                new PeriodicWorkRequest.Builder(RandomRecipeWorker.class, 1, TimeUnit.DAYS)
                        .setConstraints(constraints)
                        .build();

        WorkManager.getInstance(getApplicationContext()).enqueueUniquePeriodicWork(
                getString(R.string.random_recipe_worker_name),
                ExistingPeriodicWorkPolicy.KEEP,
                request
        );
    }
}

File: app/src/main/java/com/disruption/cookcentral/CookCentralApplication.kt
package com.disruption.cookcentral

import androidx.multidex.MultiDexApplication
import androidx.work.*
import com.disruption.cookcentral.di.appModule
import com.disruption.cookcentral.work.RandomRecipeWorker
import org.koin.android.ext.koin.androidContext
import java.util.concurrent.TimeUnit

class CookCentralApplication : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()
        startKoin()
        setUpRecipeWork()
    }

    private fun startKoin() {
        // start Koin!
        org.koin.core.context.startKoin {
            // declare used Android context
            androidContext(this@CookCentralApplication)
            // declare modules
            modules(appModule)
        }
    }

    private fun setUpRecipeWork() {
        val constraints = Constraints.Builder()
                .setRequiredNetworkType(NetworkType.CONNECTED)
                .setRequiresBatteryNotLow(true)
                .build()

        val request = PeriodicWorkRequest.Builder(RandomRecipeWorker::class.java, 1, TimeUnit.DAYS)
                .setConstraints(constraints)
                .build()

        WorkManager.getInstance(applicationContext).enqueueUniquePeriodicWork(
                getString(R.string.random_recipe_worker_name),
                ExistingPeriodicWorkPolicy.KEEP,
                request
        )
    }
}